snippets/standard/datetime/json/datetime-converter-examples/example2/Program.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Buffers;
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace DateTimeConverterExamples
+{
+    public class DateTimeConverterExample2 : JsonConverter<DateTime>
+    {
+        public override DateTime Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                throw new JsonException();
+            }
+
+            if (Utf8Parser.TryParse(reader.ValueSpan, out DateTime value, out _, 'R'))
```

Comment:
Can you verify if this works with data that needs to be un-escaped? I suspect it won't and we'd need `reader.GetString()` and then re-transcode to UTF-8.

If so, this highlights the need for even more. https://github.com/dotnet/corefx/issues/35386

Reply:
No it won't - the parser expects an un-escaped input. I've refactored this.